graphbolt/src/csc_sampling_graph.cc

```diff
@@ -97,6 +97,48 @@ void CSCSamplingGraph::Save(torch::serialize::OutputArchive& archive) const {
   }
 }
 
+void CSCSamplingGraph::SetState(
+    torch::Dict<std::string, torch::Dict<std::string, torch::Tensor>>& state) {
+  auto&& regular_tensors = state.at("regular_tensors");
+  TORCH_CHECK(
+      regular_tensors.at("magic_number").equal(torch::tensor({1})),
+      "Magic number mismatches when loading pickled CSCSamplingGraph.")
+  indptr_ = regular_tensors.at("indptr");
+  indices_ = regular_tensors.at("indices");
+  if (regular_tensors.find("node_type_offset") != regular_tensors.end()) {
+    node_type_offset_ = regular_tensors.at("node_type_offset");
+  }
+  if (regular_tensors.find("type_per_edge") != regular_tensors.end()) {
+    type_per_edge_ = regular_tensors.at("type_per_edge");
+  }
+  if (state.find("edge_attributes") != state.end()) {
+    edge_attributes_ = state.at("edge_attributes");
+  }
+}
+
+torch::Dict<std::string, torch::Dict<std::string, torch::Tensor>>
+CSCSamplingGraph::GetState() const {
+  // State is a dict of dicts. The tensor-type attributes are stored in the dict
+  // with key "regular_tensors". The dict-type attributes (edge_attributes) are
+  // stored directly with the their name as the key.
+  torch::Dict<std::string, torch::Dict<std::string, torch::Tensor>> state;
+  torch::Dict<std::string, torch::Tensor> regular_tensors;
+  regular_tensors.insert("magic_number", torch::tensor({1}));  // Magic number.
```

Comment:
Why we only need magic_number in regular_tensors?

Reply:
The magic number (or version number) is for the whole state. Only is regular_tensors a good place to put it.